core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -673,10 +666,42 @@ namespace kagome::blockchain {
   }
 
   outcome::result<void> BlockTreeImpl::markAsRevertedBlocks(
-      const std::vector<primitives::BlockInfo> &blocks) {
-    // TODO Must be implemented
-#warning "Must be implemented"
-    return outcome::success();
+      const std::vector<primitives::BlockHash> &block_hashes) {
+    return block_tree_data_.exclusiveAccess(
+        [&](auto &p) -> outcome::result<void> {
```

Comment:
```suggestion
        [&](BlockTreeData &p) -> outcome::result<void> {
```

Reply:
Why is needed to provide particular type here?

Reply:
Reduce templates, enable autocomplete and errors in editor.